From c96df29698a1ef7515cf32440d63be97f3abb965 Mon Sep 17 00:00:00 2001
From: npalanivel <npalanivel@maxlinear.com>
Date: Thu, 18 Jul 2024 12:51:44 +0530
Subject: [PATCH] WLANRTSYS-83174 : Extend check for Enterprise security modes
 for the 6GHz BSS

Porting WLANRTSYS-63986 -Fix to avoid processing CMAC-256 cipher in hapd.
Issue:
While testing the original issue in WLANRTSYS-63986, found that the
interfaces are going down if we configure CMAC-256 encryption type.
Root Cause:
Though we are processing the type in hapd, there is no code to process
the same encryption type in mtlk driver/FW.
Fix:
Removed the encryption type while processing in hapd config fill.
Todo: make feasibility study for upstream to opensource
---
 hostapd/config_file.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/hostapd/config_file.c b/hostapd/config_file.c
index 50ee5f4f5..4c12d7f78 100644
--- a/hostapd/config_file.c
+++ b/hostapd/config_file.c
@@ -4087,8 +4087,10 @@ static int hostapd_config_fill(struct hostapd_config *conf,
 			bss->group_mgmt_cipher = WPA_CIPHER_BIP_GMAC_128;
 		} else if (os_strcmp(pos, "BIP-GMAC-256") == 0) {
 			bss->group_mgmt_cipher = WPA_CIPHER_BIP_GMAC_256;
+#ifndef CONFIG_VENDOR_MXL //Currently no support for this encryption in mtlk driver/FW
 		} else if (os_strcmp(pos, "BIP-CMAC-256") == 0) {
 			bss->group_mgmt_cipher = WPA_CIPHER_BIP_CMAC_256;
+#endif /* CONFIG_VENDOR_MXL */
 		} else {
 			wpa_printf(MSG_ERROR, "Line %d: invalid group_mgmt_cipher: %s",
 				   line, pos);
-- 
2.43.0

